src/python/pants/backend/javascript/package_json.py

```diff
@@ -0,0 +1,623 @@
+
+from pants.backend.project_info import dependencies
+from pants.base.specs import AncestorGlobSpec, RawSpecs
+from pants.build_graph.address import Address
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.core.target_types import (
+    TargetGeneratorSourcesHelperSourcesField,
+    TargetGeneratorSourcesHelperTarget,
+)
+from pants.core.util_rules import stripped_source_files
+from pants.engine import fs
+from pants.engine.collection import Collection, DeduplicatedCollection
+from pants.engine.fs import DigestContents, FileContent, PathGlobs
+from pants.engine.internals import graph
+from pants.engine.internals.native_engine import Digest, Snapshot
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.rules import Rule, collect_rules, rule
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    AllTargets,
+    Dependencies,
+    DependenciesRequest,
+    GeneratedTargets,
+    GenerateTargetsRequest,
+    SequenceField,
+    SingleSourceField,
+    SourcesField,
+    StringField,
+    StringSequenceField,
+    Target,
+    TargetGenerator,
+    Targets,
+)
+from pants.engine.unions import UnionMembership, UnionRule
+from pants.option.global_options import UnmatchedBuildFileGlobs
+from pants.util.frozendict import FrozenDict
+from pants.util.strutil import softwrap
+
+
+class NodePackageDependenciesField(Dependencies):
+    pass
+
+
+class PackageJsonSourceField(SingleSourceField):
+    default = "package.json"
+    required = False
+
+
+@dataclass(frozen=True)
+class NodeBuildScript:
+    entry_point: str
+    output_directories: tuple[str, ...] = ()
+    output_files: tuple[str, ...] = ()
+    extra_caches: tuple[str, ...] = ()
+
+    @classmethod
+    def create(
+        cls,
+        entry_point: str,
+        output_directories: Iterable[str] = (),
+        output_files: Iterable[str] = (),
+        extra_caches: Iterable[str] = (),
+    ) -> NodeBuildScript:
+        return cls(
+            entry_point=entry_point,
+            output_directories=tuple(output_directories),
+            output_files=tuple(output_files),
+            extra_caches=tuple(extra_caches),
+        )
+
+
+class NodePackageScriptsField(SequenceField[NodeBuildScript]):
+    alias = "scripts"
+    expected_element_type = NodeBuildScript
+
+    help = softwrap(
+        """
+        Custom node package manager scripts that should be known
+        and ran as part of relevant goals.
+
+        Maps the package.json#scripts section to a cacheable pants invocation.
+        """
+    )
+    expected_type_description = (
+        '[node_build_script(entry_point="build", output_directories=["./dist/"], ...])'
+    )
+    default = ()
+
+
+class PackageJsonTarget(TargetGenerator):
+    alias = "package_json"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        PackageJsonSourceField,
+        NodePackageScriptsField,
+        NodePackageDependenciesField,
+    )
+    help = "A package.json file."
```

Comment:
Some help describing how `package_json` targets "own" or otherwise relate to `node_package` targets might be good... on the other hand, based on `OwningNodePackage`, it looks like it just search parent directories, which is fairly self explanatory.

This is likely one of those cases that is self explanatory if you have an existing project to run `tailor` on, but tricky to start from scratch on without more docs.